### ISSUE TITLE
oxc/only used in recursion "bad" example doesn't cause an error

### ISSUE THREAD

**Issue by camc314** [oxc-project/oxc#4811]
In the docs for oxc/only used in recursion, there's the following case:
```js
function f(a: number, b: number): number {
    if (a == 0) {
        return 1
    } else {
        return f(a - 1, b + 1)
    }
}
```

this doesn't report an error if you run this code against the lint rule 😅 

to complete this issue we should make it report the error for this case.

**Comment by DonIsaac** [comment]
I wonder if we should have our (almost added) documentation tests run the lint rule against incorrect examples 🤔

**Comment by camc314** [comment]
> I wonder if we should have our (almost added) documentation tests run the lint rule against incorrect examples 🤔

that'd be pretty sweet
